controller/server.go

```diff
@@ -20,18 +20,26 @@ func (rw *ReadWrite) Events(ctx context.Context, taskName string) (map[string][]
 }
 
 func (rw *ReadWrite) Task(ctx context.Context, taskName string) (config.TaskConfig, error) {
-	// TODO handle ctx while waiting for driver lock if it is currently active
-	d, ok := rw.drivers.Get(taskName)
-	if !ok {
-		return config.TaskConfig{}, fmt.Errorf("a task with name '%s' does not exist or has not been initialized yet", taskName)
+	// TODO handle ctx while waiting for state lock if it is currently active
+	conf, ok := rw.state.GetTask(taskName)
+	if ok {
+		return conf, nil
 	}
 
-	conf, err := configFromDriverTask(d.Task())
-	if err != nil {
-		return config.TaskConfig{}, err
+	return config.TaskConfig{}, fmt.Errorf("a task with name '%s' does not exist or has not been initialized yet", taskName)
+}
+
+func (rw *ReadWrite) Tasks(ctx context.Context) ([]config.TaskConfig, error) {
```

Comment:
Note: Tasks() returning an error is no longer necessary at this point. I have a task to revisit this at the end of refactoring to remove returning an error. I decided not to include it in this PR since its already so large

Reply:
Following up, this is addressed here https://github.com/hashicorp/consul-terraform-sync/pull/846/commits/2931a2a135d1471bd16a1735af21d50812aae9a9